File: Data Structures & Algorithms/sqrtx/submission-2.java
class Solution {
    public int mySqrt(int x) {
        int left= 0;
        int right= x;
        while(left<= right){
            long mid=(long) (left+(right-left)/2);
            long val= (mid*mid);
            if(val==x){
                return (int)mid;
            }else if(val>x){
                right= (int)mid-1;
            }else{
                left= (int)mid+1;
            }
        }
        return right;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left= 0;
        int right= heights.size()-1;
        int ans= 0;
        while(left<right){
            int height= min(heights[left], heights[right]);
            int diff= abs(right-left);
            int currRes= height*diff;
            ans= max(ans, currRes);
            if(heights[left]<heights[right]){
                left++;
            }else{
                right--;
            }
        }
        return ans;
    }
};
